module/mailnews/mailread.c

```diff
@@ -353,3 +395,153 @@ void UserMailReply(int msg_num, Bool reply_all)
    MakeReplySubject(reply->subject, MAX_SUBJECT);
    MailSendReply(hReadMailDlg, reply);
 }
+
+void OnColumnClick(LPNMLISTVIEW pLVInfo)
+{
+   int nSortColumn = pLVInfo->iSubItem;
+   bool bSortAscending = true;
+
+   if (nSortColumn == currentSortColumn)
+   {
+      // Toggle sort direction if the same column is clicked again
+      currentSortAscending = !currentSortAscending;
+   }
+   else
+   {
+      // Set ascending order for new column
+      currentSortColumn = nSortColumn;
+      currentSortAscending = true;
+   }
+
+   LPARAM lParamSort = (currentSortAscending ? 1 : -1) * (currentSortColumn + 1);
+
+   if (hbmUpArrow != NULL && hbmDownArrow != NULL)
+   {
+      ListView_SetHeaderSortImage(pLVInfo->hdr.hwndFrom, currentSortColumn, currentSortAscending);
+   }
+
+   // Sort list
+   ListView_SortItemsEx(pLVInfo->hdr.hwndFrom, CompareMailListItems, lParamSort);
```

Comment:
I want to call out that mail uses `ListView_SortItemsEx` while news uses `ListView_SortItems`. Down the line `ListView_GetItemText` (uses to retrieve text from the mail list) takes the item index.

> With ListView_SortItemsEx, the item's index is passed instead of its lparam value.

[Source](https://learn.microsoft.com/en-us/windows/win32/api/commctrl/nf-commctrl-listview_sortitemsex)